src/bus/channels.spec.js

```diff
@@ -0,0 +1,368 @@
+
+          return expect(promise).to.be.rejectedWith(
+            `A ${field} is required to create a new message bus channel.`
+          );
+        });
+      });
+    });
+  });
+
+  describe('delete', function() {
+    context('all required fields are provided', function() {
+      let channel;
+      let promise;
+
+      beforeEach(function() {
+        channel = fixture.build('channel');
+
+        const channels = new Channels(baseSdk, baseRequest);
+        channels._baseUrl = expectedHost;
+        promise = channels.delete(
+          channel.organizationId,
+          channel.serviceId,
+          channel.id
+        );
+      });
+
+      it('requests to delete the event', function() {
+        expect(baseRequest.delete).to.be.calledWith(
+          `${expectedHost}/organizations/${channel.organizationId}/services/${
+            channel.serviceId
+          }/channels/${channel.id}`
+        );
+      });
+
+      it('returns a resolved promise', function() {
+        return expect(promise).to.be.fulfilled;
+      });
+    });
+
+    context('the organizationId is not provided', function() {
+      it('throws an error', function() {
+        const channels = new Channels(baseSdk, baseRequest);
+        const promise = channels.delete();
+
+        return expect(promise).to.be.rejectedWith(
+          'An organizationId is required to delete a message bus channel.'
+        );
+      });
+    });
+
+    context('the serviceId is not provided', function() {
+      it('throws an error', function() {
+        const channels = new Channels(baseSdk, baseRequest);
+        const promise = channels.delete('1');
+
+        return expect(promise).to.be.rejectedWith(
+          'A serviceId is required to delete a message bus channel.'
+        );
+      });
+    });
+
+    context('the channelId is not provided', function() {
+      it('throws an error', function() {
+        const channels = new Channels(baseSdk, baseRequest);
+        const promise = channels.delete('1', '2');
+
+        return expect(promise).to.be.rejectedWith(
+          'A channelId is required to delete a message bus channel.'
+        );
+      });
+    });
+  });
+
+  describe('get', function() {
+    context('the required fields are provided', function() {
+      let channelFromServerAfterFormat;
+      let channelFromServerBeforeFormat;
+      let expectedOrganizationId;
+      let expectedServiceId;
+      let expectedChannelId;
+      let formatChannelFromServer;
+      let promise;
+      let request;
+
+      beforeEach(function() {
+        expectedChannelId = faker.random.uuid();
+        channelFromServerAfterFormat = fixture.build('channel', {
+          id: expectedChannelId
+        });
+        expectedOrganizationId = channelFromServerAfterFormat.organizationId;
+        expectedServiceId = channelFromServerAfterFormat.serviceId;
+        channelFromServerBeforeFormat = fixture.build(
+          'channel',
+          { id: expectedChannelId },
```

Comment:
Why not pass in channelFromServerAfterFormat?

Reply:
Ok

Reply:
You can pull `expectedChannelId` from `channelFromServerAfterFormat` too

Reply:
Ah yeah that makes it easier to read